### ISSUE TITLE
override widgetSpacing and add max lines

### ISSUE THREAD

**Issue by yahyaMajdAlamaal** [bossbeagle1509/chip_list#4]
Hi!
I'm using your package and its amazing.
but can you please add two things ASAP?

1. I want to change the widgetSpacing but I can't find it in your constructor params can you add it?
2. Can you also add a params that allow me to add maximum lines if I'm using wrapping?
**I fount a package that may help you in this**
https://pub.dev/packages/extended_wrap

**Comment by bossbeagle1509** [comment]
Thanks :)

Couple of questions:
1. You mean the spacing between the chips?
2. And maximum lines as in, you want an overflow indicator after a certain number of lines instead of rendering the entire list?

**Comment by yahyaMajdAlamaal** [comment]
Hi again

1. Yes a space between chips (I review the current code and found a property called widgetSpacing and default value for it is 4 but it's not included in constructor so I cant override it)
2. Maybe overflow is good or to make it better let developer choose overflow or maybe a callback function called onOverflow
(It's a suggestion and it will help me a lot😅)

**Comment by yahyaMajdAlamaal** [comment]
Hi again
First of all I appreciate your interest of helping, Thank you 

My use case that I want to use overflow with is that I want to show just a specific lines of chips then I add a "show more" button  to show the others.

I prefer using callback function because I want to show the others options in a bottom sheet as a list of radio or checkboxes not a chips.

Thank you.

**Comment by bossbeagle1509** [comment]
I thought about this a little bit. Instead of using fancy overflow detection (sounds like a headache), how about I enable support for lazy loading?

I can simply allow the develop to shrink or increase a buffer which is actually shown to the user. The advantage of this is, well, the lazy loading bit.

In your case, you specify definite bounds for the `ChipList`, by maybe putting it in a `SizedBox` and enable `shouldWrap`. With some changes to the API (I will try to keep it backwards compatible), you'll have full control of what you're displaying, and so you can display the rest of your options wherever you want.

TLDR, I'll add buffer support. Sounds ok?

**Comment by yahyaMajdAlamaal** [comment]
After think more I think yeah overflow detection will be a headache for you
So if you add `maxLines` property with overflow type to enable hide overflow or show it will be enough, or maybe adding `overflowWidget` to add custom widget as overflow will be prefect
Check the package I added on the question it have like this.

I tried to add `SizedBox` but it didn't work as I want

**Comment by yahyaMajdAlamaal** [comment]
Any update?

**Comment by bossbeagle1509** [comment]
Made `widgetSpacing` visible!
